drivers/timer/cortex_m_systick.c

```diff
@@ -17,7 +17,9 @@
 
 #define CYC_PER_TICK (sys_clock_hw_cycles_per_sec()	\
 		      / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
-#define MAX_TICKS ((k_ticks_t)(COUNTER_MAX / CYC_PER_TICK) - 1)
+/* add MAX_TICKS protection */
+#define _COUNTER_MAX ((COUNTER_MAX > (2 * CYC_PER_TICK)) ? COUNTER_MAX : (2 * CYC_PER_TICK))
+#define MAX_TICKS ((k_ticks_t)(_COUNTER_MAX / CYC_PER_TICK) - 1)
```

Comment:
This seems clumsy.  Can't you add a BUILD_ASSERT() in 64 bit precision instead?

Reply:
build_assert only assert during building, it does not resolve the problem, so if we want to CONFIG_SYS_CLOCK_TICKS_PER_SEC to be 100 and still running in that high frequency(960M). let me try to find a better way to implement this.